arch/arm/core/aarch32/cortex_m/fault_s.S

```diff
@@ -76,14 +76,50 @@ SECTION_SUBSEC_FUNC(TEXT,__fault,z_arm_debug_monitor)
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 SECTION_SUBSEC_FUNC(TEXT,__fault,z_arm_exc_spurious)
 
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
+	/* Build __callee_saved_esf */
+	mov r3, r11
+	mov r2, r10
+	mov r1, r9
+	mov r0, r8
+	push {r0-r3, lr}
+	push {r3-r7}
```

Comment:
shouldn't be `push {r4-r7}` ?

Reply:
good observation @ceolin  ... I tried to add a comment on line 101 about this. Per the [ARM procedure call standard ](https://documentation-service.arm.com/static/5f201281bb903e39c84d7eae), "The stack must be double-word aligned" at a public interface, so I pushed one extra register to satisfy the requirement for the call to `z_arm_fault`